src/internal/task/util.go

```diff
@@ -3,16 +3,46 @@ package task
 import (
 	"context"
 
-	taskapi "github.com/pachyderm/pachyderm/v2/src/task"
-	"go.uber.org/zap"
-
 	"github.com/gogo/protobuf/jsonpb"
 	"github.com/gogo/protobuf/types"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/sync/semaphore"
+
+	taskapi "github.com/pachyderm/pachyderm/v2/src/task"
+
 	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
 	"github.com/pachyderm/pachyderm/v2/src/internal/log"
-	"golang.org/x/sync/errgroup"
+	"github.com/pachyderm/pachyderm/v2/src/internal/taskchain"
 )
 
+// DoOrdered returns objects of type Any in order on channel out.
+func DoOrdered(ctx context.Context, doer Doer, inputs, out chan *types.Any, parallelism int) error {
+	taskChain := taskchain.NewTaskChain(ctx, semaphore.NewWeighted(int64(parallelism)))
```

Comment:
If we are going to use taskchain as the package name, then we would probably be better off with the constructor just being `New`.